core/kogitoservice/deployer.go

```diff
@@ -89,6 +89,9 @@ func NewServiceDeployer(context *operator.Context, definition ServiceDefinition,
 	if len(definition.Request.NamespacedName.Namespace) == 0 && len(definition.Request.NamespacedName.Name) == 0 {
 		panic("No Request provided for the Service Deployer")
 	}
+	if infraHandler == nil {
+		panic("InfraHandler can't be nil when creating a new ServiceDeployer")
```

Comment:
AFAIK product operator doesn't provide infraHandler currently, would crash in here.

Reply:
It was interesting because our codebase is trying to access it, and the tests were panicking, although passing. I'll double-check it. To avoid any discrepancies, I can create an `NoOp` implementation.

Reply:
That's unfortunate: https://github.com/kiegroup/rhpam-kogito-operator/blob/master/controllers/kogitoruntime_controller.go#L99

Our interface shouldn't allow this. I'll take a look.